src/containers/IntentService.jsx

```diff
@@ -36,23 +37,32 @@ export default class IntentService extends Component {
           disableSuccessTimeout: !!data.disableSuccessTimeout
         })
 
-        if (!data || !data.slug) {
+        if (!data) {
           throw new Error('Unexpected data from intent')
         }
 
-        return this.store.fetchKonnectorInfos(data.slug)
+        if (data.slug) {
+          return this.store.fetchKonnectorInfos(data.slug)
+        } else if (data.dataType) {
+          return this.store.findByDataType(data.dataType)
+        }
       })
-      .then(konnector => {
-        if (!konnector) {
+      .then(storeData => {
+        if (!storeData) {
           throw new Error(`Unknown konnector`)
+        } else if (storeData.length) {
+          this.setState({
+            isFetching: false,
+            konnectors: storeData
+          })
+        } else {
+          this.setState({
+            isFetching: false,
+            konnector: storeData
```

Comment:
We may also keep a list of konnectors, even in the case where there is only one, what do you think ?